2DPlatformerController/Assets/Characters/EnemyCreep.cs

```diff
@@ -92,8 +117,9 @@ private void Attack(IDamagable trgt, Rigidbody2D primaryCollider)
             dmgManager.DistributeDamageWithInvincible(trgt.GetVitalityAttributes(), damagableAttributes, this.damagableAttributes.AttackDamage);
         if (trgt.GetVitalityAttributes().HP <= 0)
         {
-            trgt.GetGameObject().GetComponent<SpriteRenderer>().color = Color.red;
-            //Destroy(trgt.GetGameObject());
+            //trgt.GetGameObject().GetComponent<SpriteRenderer>().color = Color.red;
+            Destroy(trgt.GetHealthSlider().gameObject);
```

Comment:
have the healthSliderManager handle destruction in its own method called DestroyHealthSlider
i.e.: you call healthSliderManager.DestroyHealthSlider(trgt.GetHealthSlider());